File: 프로그래머스/lv0/120864. 숨어있는 숫자의 덧셈 （2）/숨어있는 숫자의 덧셈 （2）.py
def solution(my_string):
    s = ''.join(i if i.isdigit() else ' ' for i in my_string)
    return sum(int(i) for i in s.split())

# 기존 풀이
#
# def solution(my_string):
#     answer = 0
#     tmp = ""
#     for i in my_string:
#         if i.isnumeric():
#             tmp += i
#         elif tmp != "":
#             answer += int(tmp)
#             tmp = ""
#     if tmp != "":
#         answer += int(tmp)
    
#     return answer
    
            

File: 프로그래머스/0/120864. 숨어있는 숫자의 덧셈 （2）/숨어있는 숫자의 덧셈 （2）.java
import java.util.*;

class Solution {
    public int solution(String my_string) {
        int answer = 0;

        String[] str = my_string.split("[a-zA-Z]");

        for(String s : str){
            if(!s.equals("")) {
                answer += Integer.parseInt(s);                
            }
        }

        return answer;
    }
}